backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotResource.java

```diff
@@ -26,14 +27,16 @@ protected BackendDiskSnapshotResource(String id, BackendDiskSnapshotsResource pa
 
     @Override
     public DiskSnapshot get() {
-        DiskSnapshot diskSnapshot = performGet(QueryType.GetDiskSnapshotByImageId,
-                new IdQueryParameters(guid), Disk.class);
+        DiskImage diskImage = runQuery(QueryType.GetDiskSnapshotByImageId,
+                new IdQueryParameters(guid)).getReturnValue();
+        DiskSnapshot diskSnapshot = map(diskImage, null);
         diskSnapshot.setDisk(new Disk());
         diskSnapshot.getDisk().setId(diskId.toString());
         diskSnapshot.getDisk().setHref(backendDiskSnapshotsResource.buildParentHref(diskId.toString(), false));
-        diskSnapshot.setHref(backendDiskSnapshotsResource.buildHref(diskId.toString(), diskSnapshot.getId().toString()));
-        if (diskSnapshot.getParent() != null) {
-            diskSnapshot.getParent().setHref(backendDiskSnapshotsResource.buildHref(diskId.toString(),
+        diskSnapshot.setHref(backendDiskSnapshotsResource.buildHref(diskId.toString(), diskSnapshot.getId()));
```

Comment:
Not related to this PR - kind of strange implementation, `buildHref()` & `buildParentHref()` can be static, thus no need in this `backendDiskSnapshotsResource` data member at all. Maybe there is a historical reason for that? :)

Reply:
I agree with @barpavel  but I am not sure if I suppose to change it in that PR.
@ahadas should I change the  BackendDiskSnapshotsResource class?

Reply:
not in this PR (as @barpavel wrote, it's unrelated to that)